From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?ZX=E5=A4=8F=E5=A4=9C=E4=B9=8B=E9=A3=8E?=
 <snwcreations@qq.com>
Date: Tue, 17 Sep 2024 20:38:03 +0800
Subject: [PATCH] refactor: removed the features that not exists in Minecraft
 1.20.1 but 1.20.4


diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/ArmadilloController.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/ArmadilloController.java.ignored
similarity index 100%
rename from v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/ArmadilloController.java
rename to v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/ArmadilloController.java.ignored
diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/BoggedController.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/BoggedController.java.ignored
similarity index 100%
rename from v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/BoggedController.java
rename to v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/BoggedController.java.ignored
diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/BreezeController.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/BreezeController.java.ignored
similarity index 100%
rename from v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/BreezeController.java
rename to v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/BreezeController.java.ignored
diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/BreezeWindChargeController.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/BreezeWindChargeController.java.ignored
similarity index 100%
rename from v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/BreezeWindChargeController.java
rename to v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/BreezeWindChargeController.java.ignored
diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/OminousItemSpawnerController.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/OminousItemSpawnerController.java.ignored
similarity index 100%
rename from v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/OminousItemSpawnerController.java
rename to v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/OminousItemSpawnerController.java.ignored
diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/WindChargeController.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/WindChargeController.java.ignored
similarity index 100%
rename from v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/WindChargeController.java
rename to v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/entity/nonliving/WindChargeController.java.ignored
diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/CustomEntityRegistry.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/CustomEntityRegistry.java
index cddac7eaddb7bc343494b8fad3b1a08649156b4e..b7e015d9978bea64c44425445050059a15308e5c 100644
--- a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/CustomEntityRegistry.java
+++ b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/CustomEntityRegistry.java
@@ -39,7 +39,7 @@ import net.minecraft.world.entity.GlowSquid;
 import net.minecraft.world.entity.Interaction;
 import net.minecraft.world.entity.LightningBolt;
 import net.minecraft.world.entity.Marker;
-import net.minecraft.world.entity.OminousItemSpawner;
+//import net.minecraft.world.entity.OminousItemSpawner; // Backport - This does not exist in MC 1.20.1
 import net.minecraft.world.entity.ambient.Bat;
 import net.minecraft.world.entity.animal.Bee;
 import net.minecraft.world.entity.animal.Cat;
@@ -65,7 +65,7 @@ import net.minecraft.world.entity.animal.TropicalFish;
 import net.minecraft.world.entity.animal.Turtle;
 import net.minecraft.world.entity.animal.Wolf;
 import net.minecraft.world.entity.animal.allay.Allay;
-import net.minecraft.world.entity.animal.armadillo.Armadillo;
+//import net.minecraft.world.entity.animal.armadillo.Armadillo; // Backport - This does not exist in MC 1.20.1
 import net.minecraft.world.entity.animal.axolotl.Axolotl;
 import net.minecraft.world.entity.animal.camel.Camel;
 import net.minecraft.world.entity.animal.frog.Frog;
@@ -91,7 +91,7 @@ import net.minecraft.world.entity.item.FallingBlockEntity;
 import net.minecraft.world.entity.item.ItemEntity;
 import net.minecraft.world.entity.item.PrimedTnt;
 import net.minecraft.world.entity.monster.Blaze;
-import net.minecraft.world.entity.monster.Bogged;
+//import net.minecraft.world.entity.monster.Bogged; // Backport - This does not exist in MC 1.20.1
 import net.minecraft.world.entity.monster.CaveSpider;
 import net.minecraft.world.entity.monster.Creeper;
 import net.minecraft.world.entity.monster.Drowned;
@@ -123,7 +123,7 @@ import net.minecraft.world.entity.monster.Zoglin;
 import net.minecraft.world.entity.monster.Zombie;
 import net.minecraft.world.entity.monster.ZombieVillager;
 import net.minecraft.world.entity.monster.ZombifiedPiglin;
-import net.minecraft.world.entity.monster.breeze.Breeze;
+//import net.minecraft.world.entity.monster.breeze.Breeze; // Backport - This does not exist in MC 1.20.1
 import net.minecraft.world.entity.monster.hoglin.Hoglin;
 import net.minecraft.world.entity.monster.piglin.Piglin;
 import net.minecraft.world.entity.monster.piglin.PiglinBrute;
@@ -148,8 +148,12 @@ import net.minecraft.world.entity.projectile.ThrownExperienceBottle;
 import net.minecraft.world.entity.projectile.ThrownPotion;
 import net.minecraft.world.entity.projectile.ThrownTrident;
 import net.minecraft.world.entity.projectile.WitherSkull;
+// Backport start - These do not exist in MC 1.20.1
+/*
 import net.minecraft.world.entity.projectile.windcharge.BreezeWindCharge;
 import net.minecraft.world.entity.projectile.windcharge.WindCharge;
+*/
+// Backport end
 import net.minecraft.world.entity.vehicle.Boat;
 import net.minecraft.world.entity.vehicle.ChestBoat;
 import net.minecraft.world.entity.vehicle.Minecart;
@@ -350,7 +354,7 @@ public class CustomEntityRegistry extends DefaultedMappedRegistry<EntityType<?>>
     static {
         minecraftClassMap.put(EntityType.ALLAY, Allay.class);
         minecraftClassMap.put(EntityType.AREA_EFFECT_CLOUD, AreaEffectCloud.class);
-        minecraftClassMap.put(EntityType.ARMADILLO, Armadillo.class);
+//        minecraftClassMap.put(EntityType.ARMADILLO, Armadillo.class); // Backport - This does not exist in MC 1.20.1
         minecraftClassMap.put(EntityType.ARMOR_STAND, ArmorStand.class);
         minecraftClassMap.put(EntityType.ARROW, Arrow.class);
         minecraftClassMap.put(EntityType.AXOLOTL, Axolotl.class);
@@ -359,9 +363,13 @@ public class CustomEntityRegistry extends DefaultedMappedRegistry<EntityType<?>>
         minecraftClassMap.put(EntityType.BLAZE, Blaze.class);
         minecraftClassMap.put(EntityType.BLOCK_DISPLAY, Display.BlockDisplay.class);
         minecraftClassMap.put(EntityType.BOAT, Boat.class);
+        // Backport start - These do not exist in MC 1.20.1
+        /*
         minecraftClassMap.put(EntityType.BOGGED, Bogged.class);
         minecraftClassMap.put(EntityType.BREEZE, Breeze.class);
         minecraftClassMap.put(EntityType.BREEZE_WIND_CHARGE, BreezeWindCharge.class);
+        */
+        // Backport end
         minecraftClassMap.put(EntityType.CAMEL, Camel.class);
         minecraftClassMap.put(EntityType.CAT, Cat.class);
         minecraftClassMap.put(EntityType.CAVE_SPIDER, CaveSpider.class);
@@ -409,7 +417,7 @@ public class CustomEntityRegistry extends DefaultedMappedRegistry<EntityType<?>>
         minecraftClassMap.put(EntityType.ITEM, ItemEntity.class);
         minecraftClassMap.put(EntityType.ITEM_DISPLAY, Display.ItemDisplay.class);
         minecraftClassMap.put(EntityType.ITEM_FRAME, ItemFrame.class);
-        minecraftClassMap.put(EntityType.OMINOUS_ITEM_SPAWNER, OminousItemSpawner.class);
+//        minecraftClassMap.put(EntityType.OMINOUS_ITEM_SPAWNER, OminousItemSpawner.class); // Backport - This does not exist in MC 1.20.1
         minecraftClassMap.put(EntityType.FIREBALL, LargeFireball.class);
         minecraftClassMap.put(EntityType.LEASH_KNOT, LeashFenceKnotEntity.class);
         minecraftClassMap.put(EntityType.LIGHTNING_BOLT, LightningBolt.class);
@@ -465,7 +473,7 @@ public class CustomEntityRegistry extends DefaultedMappedRegistry<EntityType<?>>
         minecraftClassMap.put(EntityType.VINDICATOR, Vindicator.class);
         minecraftClassMap.put(EntityType.WANDERING_TRADER, WanderingTrader.class);
         minecraftClassMap.put(EntityType.WARDEN, Warden.class);
-        minecraftClassMap.put(EntityType.WIND_CHARGE, WindCharge.class);
+//        minecraftClassMap.put(EntityType.WIND_CHARGE, WindCharge.class); // Backport - This does not exist in MC 1.20.1
         minecraftClassMap.put(EntityType.WITCH, Witch.class);
         minecraftClassMap.put(EntityType.WITHER, WitherBoss.class);
         minecraftClassMap.put(EntityType.WITHER_SKELETON, WitherSkeleton.class);
diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/EntityPathfindingContext.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/EntityPathfindingContext.java.ignored
similarity index 100%
rename from v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/EntityPathfindingContext.java
rename to v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/EntityPathfindingContext.java.ignored
diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
index 7e8a59aa61e6ab9aa31b5001f5c9c641258d6217..036d881799f60ccff17501499ef56d51d8f96904 100644
--- a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
+++ b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
@@ -86,14 +86,18 @@ import net.citizensnpcs.api.util.EntityDim;
 import net.citizensnpcs.api.util.Messaging;
 import net.citizensnpcs.api.util.SpigotUtil.InventoryViewAPI;
 import net.citizensnpcs.nms.v1_20_R4.entity.AllayController;
-import net.citizensnpcs.nms.v1_20_R4.entity.ArmadilloController;
+//import net.citizensnpcs.nms.v1_20_R4.entity.ArmadilloController; // Backport - This does not exist in MC 1.20.1
 import net.citizensnpcs.nms.v1_20_R4.entity.ArmorStandController;
 import net.citizensnpcs.nms.v1_20_R4.entity.AxolotlController;
 import net.citizensnpcs.nms.v1_20_R4.entity.BatController;
 import net.citizensnpcs.nms.v1_20_R4.entity.BeeController;
 import net.citizensnpcs.nms.v1_20_R4.entity.BlazeController;
+// Backport start - These do not exist in MC 1.20.1
+/*
 import net.citizensnpcs.nms.v1_20_R4.entity.BoggedController;
 import net.citizensnpcs.nms.v1_20_R4.entity.BreezeController;
+*/
+// Backport end
 import net.citizensnpcs.nms.v1_20_R4.entity.CamelController;
 import net.citizensnpcs.nms.v1_20_R4.entity.CatController;
 import net.citizensnpcs.nms.v1_20_R4.entity.CaveSpiderController;
@@ -172,7 +176,7 @@ import net.citizensnpcs.nms.v1_20_R4.entity.ZombieVillagerController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.AreaEffectCloudController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.BlockDisplayController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.BoatController;
-import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.BreezeWindChargeController;
+//import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.BreezeWindChargeController; // Backport - This does not exist in MC 1.20.1
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.ChestBoatController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.DragonFireballController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.EggController;
@@ -199,7 +203,7 @@ import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.MinecartFurnaceController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.MinecartHopperController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.MinecartRideableController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.MinecartTNTController;
-import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.OminousItemSpawnerController;
+//import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.OminousItemSpawnerController; // Backport - This does not exist in MC 1.20.1
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.PaintingController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.ShulkerBulletController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.SmallFireballController;
@@ -211,7 +215,7 @@ import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.ThrownExpBottleController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.ThrownPotionController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.ThrownTridentController;
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.TippedArrowController;
-import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.WindChargeController;
+//import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.WindChargeController; // Backport - This does not exist in MC 1.20.1
 import net.citizensnpcs.nms.v1_20_R4.entity.nonliving.WitherSkullController;
 import net.citizensnpcs.npc.EntityControllers;
 import net.citizensnpcs.npc.ai.MCNavigationStrategy.MCNavigator;
@@ -336,7 +340,7 @@ import net.minecraft.world.entity.animal.Pufferfish;
 import net.minecraft.world.entity.animal.Rabbit;
 import net.minecraft.world.entity.animal.Turtle;
 import net.minecraft.world.entity.animal.allay.Allay;
-import net.minecraft.world.entity.animal.armadillo.Armadillo;
+//import net.minecraft.world.entity.animal.armadillo.Armadillo; // Backport - This does not exist in MC 1.20.1
 import net.minecraft.world.entity.animal.axolotl.Axolotl;
 import net.minecraft.world.entity.animal.camel.Camel;
 import net.minecraft.world.entity.animal.horse.AbstractHorse;
@@ -955,7 +959,7 @@ public class NMSImpl implements NMSBridge {
         registerTraitWithCommand(manager, EnderDragonTrait.class);
         registerTraitWithCommand(manager, AllayTrait.class);
         registerTraitWithCommand(manager, AxolotlTrait.class);
-        registerTraitWithCommand(manager, ArmadilloTrait.class);
+//        registerTraitWithCommand(manager, ArmadilloTrait.class); // Backport - Armadillo does not exist in MC 1.20.1
         registerTraitWithCommand(manager, BeeTrait.class);
         registerTraitWithCommand(manager, BossBarTrait.class);
         registerTraitWithCommand(manager, CamelTrait.class);
@@ -984,7 +988,7 @@ public class NMSImpl implements NMSBridge {
 
         EntityControllers.setEntityControllerForType(EntityType.ALLAY, AllayController.class);
         EntityControllers.setEntityControllerForType(EntityType.AREA_EFFECT_CLOUD, AreaEffectCloudController.class);
-        EntityControllers.setEntityControllerForType(EntityType.ARMADILLO, ArmadilloController.class);
+//        EntityControllers.setEntityControllerForType(EntityType.ARMADILLO, ArmadilloController.class); // Backport - This does not exist in MC 1.20.1
         EntityControllers.setEntityControllerForType(EntityType.ARMOR_STAND, ArmorStandController.class);
         EntityControllers.setEntityControllerForType(EntityType.ARROW, TippedArrowController.class);
         EntityControllers.setEntityControllerForType(EntityType.AXOLOTL, AxolotlController.class);
@@ -993,9 +997,13 @@ public class NMSImpl implements NMSBridge {
         EntityControllers.setEntityControllerForType(EntityType.BLAZE, BlazeController.class);
         EntityControllers.setEntityControllerForType(EntityType.BLOCK_DISPLAY, BlockDisplayController.class);
         EntityControllers.setEntityControllerForType(EntityType.BOAT, BoatController.class);
+        // Backport start - These do not exist in MC 1.20.1
+        /*
         EntityControllers.setEntityControllerForType(EntityType.BOGGED, BoggedController.class);
         EntityControllers.setEntityControllerForType(EntityType.BREEZE, BreezeController.class);
         EntityControllers.setEntityControllerForType(EntityType.BREEZE_WIND_CHARGE, BreezeWindChargeController.class);
+        */
+        // Backport end
         EntityControllers.setEntityControllerForType(EntityType.CAMEL, CamelController.class);
         EntityControllers.setEntityControllerForType(EntityType.CAT, CatController.class);
         EntityControllers.setEntityControllerForType(EntityType.CAVE_SPIDER, CaveSpiderController.class);
@@ -1046,8 +1054,12 @@ public class NMSImpl implements NMSBridge {
         EntityControllers.setEntityControllerForType(EntityType.ITEM, ItemController.class);
         EntityControllers.setEntityControllerForType(EntityType.ITEM_DISPLAY, ItemDisplayController.class);
         EntityControllers.setEntityControllerForType(EntityType.ITEM_FRAME, ItemFrameController.class);
+        // Backport start - This does not exist in MC 1.20.1
+        /*
         EntityControllers.setEntityControllerForType(EntityType.OMINOUS_ITEM_SPAWNER,
                 OminousItemSpawnerController.class);
+        */
+        // Backport end
         EntityControllers.setEntityControllerForType(EntityType.LEASH_KNOT, LeashController.class);
         EntityControllers.setEntityControllerForType(EntityType.LLAMA, LlamaController.class);
         EntityControllers.setEntityControllerForType(EntityType.LLAMA_SPIT, LlamaSpitController.class);
@@ -1102,7 +1114,7 @@ public class NMSImpl implements NMSBridge {
         EntityControllers.setEntityControllerForType(EntityType.VINDICATOR, VindicatorController.class);
         EntityControllers.setEntityControllerForType(EntityType.WANDERING_TRADER, WanderingTraderController.class);
         EntityControllers.setEntityControllerForType(EntityType.WARDEN, WardenController.class);
-        EntityControllers.setEntityControllerForType(EntityType.WIND_CHARGE, WindChargeController.class);
+//        EntityControllers.setEntityControllerForType(EntityType.WIND_CHARGE, WindChargeController.class); // Backport - This does not exist in MC 1.20.1
         EntityControllers.setEntityControllerForType(EntityType.WITCH, WitchController.class);
         EntityControllers.setEntityControllerForType(EntityType.WITHER, WitherController.class);
         EntityControllers.setEntityControllerForType(EntityType.WITHER_SKELETON, SkeletonWitherController.class);
@@ -1483,6 +1495,8 @@ public class NMSImpl implements NMSBridge {
 
     @Override
     public void setArmadilloState(org.bukkit.entity.Entity entity, ArmadilloState state) {
+        // Backport - Armadillo does not exist in MC 1.20.1
+        /*
         Armadillo.ArmadilloState s = Armadillo.ArmadilloState.IDLE;
         switch (state) {
             case IDLE:
@@ -1499,6 +1513,8 @@ public class NMSImpl implements NMSBridge {
                 break;
         }
         ((Armadillo) getHandle(entity)).switchToState(s);
+        */
+        // Backport end
     }
 
     @Override
@@ -2624,7 +2640,7 @@ public class NMSImpl implements NMSBridge {
         }
     }
 
-    private static final MethodHandle ARMADILLO_SCUTE_TIME = NMS.getSetter(Armadillo.class, "cj");
+    private static final MethodHandle ARMADILLO_SCUTE_TIME = null /*NMS.getSetter(Armadillo.class, "cj")*/; // Backport - Armadillo does not exist in MC 1.20.1
 
     private static final MethodHandle ATTRIBUTE_PROVIDER_MAP = NMS.getFirstGetter(AttributeSupplier.class, Map.class);
 
