From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?ZX=E5=A4=8F=E5=A4=9C=E4=B9=8B=E9=A3=8E?=
 <snwcreations@qq.com>
Date: Thu, 19 Sep 2024 11:37:53 +0800
Subject: [PATCH] fix: backport implementation of NMS#getRidingHeightOffset for
 Minecraft 1.20.1


diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
index 27c3bd951accfae797155caceb22f8d581364f77..825b67f317c74b8b6a555f2fcba920d9184d039f 100644
--- a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
+++ b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
@@ -774,8 +774,13 @@ public class NMSImpl implements NMSBridge {
 
     @Override
     public float getRidingHeightOffset(org.bukkit.entity.Entity entity, org.bukkit.entity.Entity mount) {
+        // Backport start - Re-implement getRidingHeightOffset using MC 1.20.1 method
+        /*
         Entity handle = getHandle(entity);
         return (float) (handle.getPassengerRidingPosition(getHandle(mount)).y - handle.position().y);
+        */
+        return (float) getHandle(mount).getPassengersRidingOffset();
+        // Backport end
     }
 
     @Override
