From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?ZX=E5=A4=8F=E5=A4=9C=E4=B9=8B=E9=A3=8E?=
 <snwcreations@qq.com>
Date: Sat, 21 Sep 2024 21:28:05 +0800
Subject: [PATCH] fix: different obfuscated field/method name causes the plugin
 refuse to work


diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
index 830a2f3c99e0bfd5b98bb9ea0d1fca0a99a4c770..42e646801c884ddab788e0854d3d0aa7409fb464 100644
--- a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
+++ b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
@@ -2798,10 +2798,10 @@ public class NMSImpl implements NMSBridge {
     private static final MethodHandle FLYING_MOVECONTROL_FLOAT_SETTER = NMS.getFirstSetter(FlyingMoveControl.class,
             boolean.class);
     private static final Location FROM_LOCATION = new Location(null, 0, 0, 0);
-    private static final MethodHandle HEAD_HEIGHT = NMS.getSetter(Entity.class, "bg");
+    private static final MethodHandle HEAD_HEIGHT = NMS.getSetter(Entity.class, "bi"); // Backport - Update obfuscated name
     private static EntityDataAccessor<Float> INTERACTION_HEIGHT = null;
     private static EntityDataAccessor<Float> INTERACTION_WIDTH = null;
-    private static final MethodHandle JUMP_FIELD = NMS.getGetter(LivingEntity.class, "bn");
+    private static final MethodHandle JUMP_FIELD = NMS.getGetter(LivingEntity.class, "bj"); // Backport - Update obfuscated name
     private static final MethodHandle LOOK_CONTROL_SETTER = NMS.getFirstSetter(Mob.class, LookControl.class);
     private static MethodHandle META_COMPOUND_TAG;
     private static final MethodHandle MINECRAFT_CLIENT = NMS.getFirstGetter(YggdrasilMinecraftSessionService.class,
@@ -2817,12 +2817,16 @@ public class NMSImpl implements NMSBridge {
             .getFirstFinalSetter(ClientboundPlayerInfoUpdatePacket.class, List.class);
 
     private static final MethodHandle PLAYERINFO_ENTRIES = PLAYER_INFO_ENTRIES_LIST;
-    private static final MethodHandle PORTAL_ENTRANCE_POS_GETTER = NMS.getGetter(Entity.class, "ay");
-    private static final MethodHandle PORTAL_ENTRANCE_POS_SETTER = NMS.getSetter(Entity.class, "ay");
+    // Backport start - Update obfuscated name
+    private static final MethodHandle PORTAL_ENTRANCE_POS_GETTER = NMS.getGetter(Entity.class, "ax");
+    private static final MethodHandle PORTAL_ENTRANCE_POS_SETTER = NMS.getSetter(Entity.class, "ax");
+    // Backport end
     private static final MethodHandle POSITION_CODEC_GETTER = NMS.getFirstGetter(ServerEntity.class,
             VecDeltaCodec.class);
-    private static final MethodHandle PUFFERFISH_DEFLATE = NMS.getSetter(Pufferfish.class, "bZ");
-    private static final MethodHandle PUFFERFISH_INFLATE = NMS.getSetter(Pufferfish.class, "bY");
+    // Backport start - Update obfuscated name
+    private static final MethodHandle PUFFERFISH_DEFLATE = NMS.getSetter(Pufferfish.class, "bU");
+    private static final MethodHandle PUFFERFISH_INFLATE = NMS.getSetter(Pufferfish.class, "bT");
+    // Backport end
     public static EntityDataAccessor<Integer> RABBIT_TYPE_DATAWATCHER = null;
     private static final Random RANDOM = Util.getFastRandom();
     private static final MethodHandle SERVER_ENTITY_GETTER = NMS.getFirstGetter(TrackedEntity.class,
@@ -2843,7 +2847,7 @@ public class NMSImpl implements NMSBridge {
         }
         try {
             // Middle one
-            ENDERMAN_CREEPY = (EntityDataAccessor<Boolean>) NMS.getField(EnderMan.class, "ca").get(null);
+            ENDERMAN_CREEPY = (EntityDataAccessor<Boolean>) NMS.getField(EnderMan.class, "bV").get(null); // Backport - Update obfuscated name
         } catch (Exception e) {
         }
         try {
@@ -2860,8 +2864,10 @@ public class NMSImpl implements NMSBridge {
         } catch (Throwable e) {
         }
         try {
-            DATA_POSE = (EntityDataAccessor<Pose>) NMS.getGetter(Entity.class, "at").invoke();
-            DATA_NAME_VISIBLE = (EntityDataAccessor<Boolean>) NMS.getGetter(Entity.class, "aT").invoke();
+            // Backport start - Update obfuscated name
+            DATA_POSE = (EntityDataAccessor<Pose>) NMS.getGetter(Entity.class, "as").invoke();
+            DATA_NAME_VISIBLE = (EntityDataAccessor<Boolean>) NMS.getGetter(Entity.class, "aV").invoke();
+            // Backport end
         } catch (Throwable e) {
         }
     }
